Changelogs for Digital Twin Manager Microservice 1024.2.0 (#4332)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/dtm-1024-2-0-added-more-details-to-error-message-when-trying-to-create-an-asset-definition-on-a-subtenant-of-an-enterprise-tenant.md

---
date: ""

> title: "Improved error feedback when trying to create an Asset Definition on a subtenant of an enterprise tenant using the Assets REST API"

Reviewer: ```suggestion
title: "Improved error message returned from the REST API on creating or updating asset definitions from a subtenant
```

product_area: "Application enablement & solutions"
change_type:
    - value: "change-2c7RdTdXo4"
      label: "Improvement"
component:
    - value: "component-Tl88RYb4A"
      label: "Digital Twin Manager"
build_artifact:
    - value: "tc-wYIY0MBDO"
      label: "dtm"
ticket: "CTM-2740"
version: "1024.2.0"
---

> Previously, the error message returned from the RestAPI when creating or updating asset definitions on a subtenant of an enterprise tenant was generic and did not include tenant IDs. The error message now includes the relevant tenant IDs.

Reviewer: ```suggestion
Previously, the error message returned from the REST API when trying to create or update asset definitions on a subtenant of an {{< enterprise-tenant >}} was generic and did not include tenant IDs. The error message now includes the relevant tenant IDs.
```

> Previously, the error message returned from the RestAPI when creating or updating asset definitions on a subtenant of an enterprise tenant was generic and did not include tenant IDs. The error message now includes the relevant tenant IDs.

Reviewer: What is the error?

Reviewer 2: > What is the error?

"Access Denied. Tenant %s has no write access because it is a subtenant of the enterprise tenant %s."

---
date: ""

> title: "Added more details to error message when trying to create an Asset Definition on a subtenant of an enterprise tenant"

Reviewer 3: ```suggestion
title: "Improved error feedback when trying to create an Asset Definition on a subtenant of an enterprise tenant using the Assets REST API"
```

product_area: "Application enablement & solutions"
change_type:
    - value: "change-2c7RdTdXo4"
      label: "Improvement"
component:
    - value: "component-Tl88RYb4A"
      label: "Digital Twin Manager"
build_artifact:
    - value: "tc-wYIY0MBDO"
      label: "dtm"
ticket: "CTM-2740"
version: "1024.2.0"
---

> More details had been added to the error message of the REST API

Reviewer 4: Missing what was the issue originally

Reviewer 3: @mvoigt-sag, can you please add what was missing previously here?

Reviewer 2: > @mvoigt-sag, can you please add what was missing previously here?

pls see my suggestion

Reviewer 3: Modified the suggestion from Michael for the changelogs and updated the file

responses when trying to create or update Asset Definitions on a

> subtenant of an enterprise tenant.

Reviewer 2: ```suggestion
Previously, the error message of the REST API when trying to create or update Asset Definitions on a
subtenant of an enterprise tenant was quite generic and lacked information about the tenant ids. This fix improves the error message.
```

> subtenant of an enterprise tenant.

Reviewer 3: ```suggestion
Previously, the error message returned from the RestAPI when creating or updating asset definitions on a subtenant of an enterprise tenant was generic and did not include tenant IDs. The error message now includes the relevant tenant IDs.
```
